api/src/main/java/io/druid/data/input/FirehoseFactory.java

```diff
@@ -36,7 +37,13 @@
    * If this method returns null, then any attempt to call hasMore(), nextRow(), commit() and close() on the return
    * value will throw a surprising NPE.   Throwing IOException on connection failure or runtime exception on
    * invalid configuration is preferred over returning null.
+   * <p/>
+   * Some fire hoses like {@link PrefetchableTextFilesFirehoseFactory} may use a temporary
+   * directory to cache data in it.
+   *
+   * @param parser             an input row parser
+   * @param temporaryDirectory a directory where temporary files are stored
    */
-  public Firehose connect(T parser) throws IOException, ParseException;
+  Firehose connect(T parser, File temporaryDirectory) throws IOException, ParseException;
```

Comment:
I realized that FirehoseFactory is in the druid api module. I think it would be better to manage temporaryDirectory internally rather than leaving the management to users.